adhocteam/Head-Start-TTADP
# Grantees from eclkc api data

**Description of change**
An an ability to query the eclkc api data to retrieve grantee names and grant numbers. In particular, this PR allows querying the programs and centers endpoints through eclkc. New endpoints in ttasmarthub accept query parameters and pass the requests to the eclkc server. The returned data is modified to include the most relevant information. 

**How to test**
- Locally, update the `.env` file with `ECLKC_API_KEY=<key>`
- `yarn deps`
- `yarn start server`
- navigate to ` localhost:8080/v1/programs?region=1&type=1`
- after login in, verify that data including grantee name and grant number(s) is returned
- pick a grant number and navigate to` localhost:8080/v1/centers?grantNumber=<grant_number>` e.g. 01CH011480
- verify that center names along with delegate numbers and grant numbers are displayed

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/38

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
